src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodec.java

```diff
@@ -167,6 +167,24 @@ static <JavaT> ToCQL<JavaT, JavaT> unsafeIdentity() {
       return (toCQLType, value) -> value;
     }
 
+    /**
+     * Returns given String if (and only if) it only contains 7-bit ASCII characters; otherwise it
+     * will throw an {@link ToCQLCodecException}
+     */
+    static String safeAscii(DataType targetTextType, String value) throws ToCQLCodecException {
+      for (int i = 0, len = value.length(); i < len; ++i) {
+        if (value.charAt(i) > 0x7F) {
```

Comment:
see comments for JSONCodecRegistryTestData

Reply:
Will add comment stating why this should be fine.

Reply:
Added.